tools/PublishModules.ps1

```diff
@@ -268,8 +276,9 @@ function Publish-RMModule
         $moduleName = (Get-Item -Path $Path).Name
         $moduleManifest = $moduleName + ".psd1"
         $moduleSourcePath = Join-Path -Path $Path -ChildPath $moduleManifest
-        $manifest = Test-ModuleManifest -Path $moduleSourcePath
-        $nupkgPath = Join-Path -Path $TempRepoPath -ChildPath ($moduleName + "." + $manifest.Version.ToString() + ".nupkg")
+        $file = Get-Item $moduleSourcePath
+        Import-LocalizedData -BindingVariable ModuleMetadata -BaseDirectory $file.DirectoryName -FileName $file.Name
+        $nupkgPath = Join-Path -Path $TempRepoPath -ChildPath ($moduleName + "." + $ModuleMetadata.ModuleVersion.ToString() + ".nupkg")
```

Comment:
Before we call PublishModule, we should add in 'ExternalModuleDependencies', and we should remove it before the final nuget package refactoring, liek here:  https://github.com/markcowl/azure-powershell/blob/fixazs/tools/PublishModules.ps1#L283